src/compiler/checker.ts

```diff
@@ -29679,9 +29679,6 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         if (!canCollectSymbolAliasAccessabilityData) {
             return;
         }
-        if (location.flags & NodeFlags.Ambient) {

```

Comment:
Yeah, I wouldn't want to remove this outright - perf tests won't show it, but this is pretty important to keep this fast from my testing. I'd just except ambient property signatures from the early skip, since they're like... not really ambient? It looks like when we have something like
```ts
class A {
  @decorator(class B {})
  declare prop: string;
}
```
the `prop` is ambient marked, but the decorator is not, so we should just be able to except the prop, like @rbuckton says.

```
if (location.flags & NodeFlags.Ambient && !isPropertySignature(location) && !isPropertyDeclaration(location)) {
```
seems like it aughta work?

Reply:
`declare` fields aren't *quite* as ambient as other declarations. They were added as a compatiblity mechanism due to native class fields using Define semantics, with legacy decorators being one of the main motivations.